roachtest: unskip synctest

nicktrav opened the pull request:
Unskip a long skipped roachtest.

Release note: None.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/88312)
<!-- Reviewable:end -->

nicktrav committed:
kvtenantccl: unskip TestTenantUpgrade

This test was skipped as part of #97076. Re-enable the test.

Fix #97076.

Release note: None.
